src/components/common/TimeZoneSelectorWithCheckbox.tsx

```diff
@@ -0,0 +1,64 @@
+import { Checkbox } from '@terraware/web-components';
+import { useEffect, useState } from 'react';
+import strings from 'src/strings';
+import { TimeZoneDescription } from 'src/types/TimeZones';
+import { useDefaultTimeZone } from 'src/utils/useTimeZoneUtils';
+import TimeZoneSelector from '../TimeZoneSelector';
+
+type UpdateTimeZone = {
+  timeZone?: string;
+};
+
+interface TimeZoneSelectorWithCheckboxProps<T extends UpdateTimeZone> {
+  onChangeTimeZone: (tzSelected: TimeZoneDescription) => void;
+  record: T;
+  setRecord: React.Dispatch<React.SetStateAction<T>>;
+}
+
+export default function TimeZoneSelectorWithCheckbox<T extends UpdateTimeZone>(
+  props: TimeZoneSelectorWithCheckboxProps<T>
+): JSX.Element {
+  const [orgTZChecked, setOrgTZChecked] = useState<boolean>(false);
+  const { onChangeTimeZone, record, setRecord } = props;
```

Comment:
we don't need setRecord,, we can just maintain a local state.. also might end up modifying the parent record if user cancels
```
  const [timeZoneEntity, setTimeZoneEntity] = useState<TimeZoneEntity>(location);
```

Reply:
Thanks!